src/routes/jira/configuration/jira-configuration-get.ts

```diff
@@ -94,42 +209,11 @@ export const JiraConfigurationGet = async (
 
 		req.log.debug("Received jira configuration page request");
 
-		const subscriptions = await Subscription.getAllForHost(jiraHost);
-		const installations = await getInstallations(subscriptions, req.log, gitHubAppId);
-
-		const failedConnections: FailedConnection[] = await Promise.all(
-			installations.rejected.map(async (installation) => {
-				const sub = subscriptions.find((sub: Subscription) => installation.id === sub.gitHubInstallationId);
-				const repo = sub && await RepoSyncState.findOneFromSubscription(sub);
-				return {
-					id: installation.id,
-					deleted: installation.deleted,
-					orgName: repo?.repoOwner
-				};
-			}));
-
-		const successfulConnections: SuccessfulConnection[] = installations.fulfilled
-			.map((installation) => ({
-				...installation,
-				isGlobalInstall: installation.repository_selection === "all"
-			}));
-
-		const handleNavigationClassName = await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost)
-			? "select-github-version-link"
-			: "add-organization-link";
-
-		res.render("jira-configuration.hbs", {
-			host: jiraHost,
-			successfulConnections,
-			failedConnections,
-			hasConnections: !!installations.total,
-			APP_URL: process.env.APP_URL,
-			csrfToken: req.csrfToken(),
-			nonce: res.locals.nonce,
-			handleNavigationClassName,
-			isGitHubCloudApp: await isGitHubCloudApp(gitHubAppId)
-		});
-
+		if (await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost)) {
+			res.render("jira-configuration-new.hbs", await JiraCloudAndEnterpriseConfiguration(res, req));
+		} else {
+			res.render("jira-configuration.hbs", await JiraCloudConfiguration(res, req));
+		}
```

Comment:
this is a weird pattern.  The `res.render` should be within each function, not outside of it.  All this conditional is doing is a custom routing based on an FF, that's it.  

A good test of this is if I were to create a new route (let's say `/jira/configuration/new` and add `JiraCloudAndEnterpriseConfiguration` to be the handler, it should render like it should by itself.  Right now, I don't think it would do anything.

Reply:
also, btw, adding that path might not be a bad idea to test stuff in prod without having the FF turned on :)

Reply:
I'm not sure what you mean by `each function`. 
As discussed earlier in this PR, the FF separates the existing view with the new one. And I do think using FF is a better approach than using a new separate route. That way we can turn on the new flow for any _Jira instance_ by turning on the FF, without changing any code.

Reply:
I think what @mboudreau is saying is that you should be calling res.render from within JiraCloudAndEnterpriseConfiguration and JiraCloudConfiguration. You can still use feature flags to do this. For example, instead of having `res.render("jira-configuration-new.hbs", await JiraCloudAndEnterpriseConfiguration(res, req))` inside the FF check just call the function `await JiraCloudAndEnterpriseConfiguration(res, req)` (although I'd probably rename it to something like `jiraCloudAndEnterpriseServer` or ` renderJiraCloudAndEnterpriseServerView`) then inside that function have 

```
	res.render("jira-configuration-new.hbs", {
		host: jiraHost,
		gheServers: groupedGheServers,
		ghCloud: { successfulConnections, failedConnections },
		hasCloudAndEnterpriseServers: !!((successfulConnections.length || failedConnections.length) && gheServers.length),
		hasCloudServers: !!(successfulConnections.length || failedConnections.length),
		hasConnections: !!(installations.total || gheServers?.length),
		APP_URL: process.env.APP_URL,
		csrfToken: req.csrfToken(),
		nonce
	});
```

and the conditional check for the FF could be simplified to something like:
```
await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost)
			? renderJiraCloudAndEnterpriseServerView(res, req)
			: renderJiraCloudView(res, req);
```

Reply:
Updated